src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java

```diff
@@ -83,7 +86,7 @@ public static List<ModuleDetails> readAllModulesIncludingThirdPartyIfAny(
             allMetadataSources.addAll(thirdPartyModuleFileNames);
         }
 
-        final List<ModuleDetails> result = new ArrayList<>(allMetadataSources.size());
+        final Map<String, ModuleDetails> result = new HashMap<>(allMetadataSources.size());
```

Comment:
@nick-mancuso What is your opinion on this change? I understand why it was made, but I am not sure it is warranted.

Reply:
>This class is used by plugins that need load of metadata from XML files.

We cannot change API of this utility if plugins depend on it. However, I would consider allowing class to be extended for our purposes. @rnveach thoughts?

Reply:
https://github.com/checkstyle/eclipse-cs/blob/f877da5251493164033ee65cf71a19dd9904d533/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java#L582
Eclipse-cs is the only registered user right now as I know. Sevntu was attempting to integrate this but has been paused. So public method can't be changed.

I believe Extending class is ok, but be this is an open issue to change the signature some, so it could impact this test some. Also just know this method call isn't a read only, it actually writes files. I know this because I created an open issue about it. Also if we are extending, I would want more than just wrapping the existing public. If we are going to wrap the existing public, then really the code could just be in this test.